app/utils/signed_data.py

```diff
@@ -0,0 +1,28 @@
+from flask import current_app
+from flask.sessions import SecureCookieSessionInterface
+from itsdangerous import BadSignature, SignatureExpired
+
+# can access for 30 days
+MAX_AGE_SECONDS = 60 * 60 * 24 * 30
+
+
+def sign_service_and_document_id(service_id, document_id):
+    signer = SecureCookieSessionInterface().get_signing_serializer(current_app)
+    return signer.dumps(
+        {
+            "service_id": service_id,
+            "document_id": document_id,
+        }
+    )
+
+
+def verify_signed_service_and_document_id(signed_data, expected_service_id, expected_document_id):
+    signer = SecureCookieSessionInterface().get_signing_serializer(current_app)
+    try:
+        data = signer.loads(signed_data, max_age=MAX_AGE_SECONDS)
+        return data == {
+            "service_id": str(expected_service_id),
+            "document_id": str(expected_document_id),
+        }
```

Comment:
This bit doesn't work for me locally - the data has UUIDs in but we're comparing it against forced str() uuids here. Are you seeing different behaviour somehow?

Reply:
@samuelhwilliams good point! i think i confused myself this morning with some unit test fixture stuff. it does indeed need to be a `UUID`. i've updated to remove the `str` and fixed the unit test.